### PR TITLE
serialization/deserialization

### PR DESCRIPTION
**Pull request by Cealtea**
added serialization/deserialization trie data.

### CODE REVIEW

**Comment by eugene-eeo** [comment]
Thanks for the PR! Minor nitpick- in the documentation for the `add` method, can we change the wording for the `data` parameter? I think it confuses people. However I'm open to discussion so yeah.

**Comment by Cealtea** [comment]
done!

File: octopi.js [hunk 1]
```diff
@@ -35,4 +43,29 @@ Octopi.prototype = {
...
     });
   }
+  /**
+     * Serialize the Octopi trie as string
+     *
+     * @return String
+     */
+  serialize: function(){
+    
+    var o = new Object();
```
Comment (eugene-eeo) [inline octopi.js:53]: Hey, any reason why this is used instead of an object literal?

File: octopi.js [hunk 2]
```diff
@@ -35,4 +43,29 @@ Octopi.prototype = {
...
+    o.table = this.table;
+
+    return JSON.stringify(o);
+
+  },  
+  /**
+     * Take serialized Octopi data as string and initialize the Octopi object
+     * @param json String The serialized Octopi trie
+     */
+  load: function(json){
```
Comment (eugene-eeo) [inline octopi.js:65]: Also I think method should be added as a "class method" instead of a method on the instance:

``` js
Octopi.load = function(json) {
  // ...
}
```

File: octopi.js (hunk 1 above)
Comment (Cealtea) [inline octopi.js:53]: No particular reason! Feel free to make some changes :)

## 

## Celine Bursztein

http://celine.im
@cealtea https://twitter.com/cealtea on Twitter, +Celine Bursztein
https://plus.google.com/+Celine%20Bursztein on Google+

**Comment by eugene-eeo** [comment]
@Cealtea  Sorry, reverting the merge until some issues are fixed. Also if you have time please add some tests for the expected behavior.

**Comment by Cealtea** [comment]
I don't think I will have more time to work on octopi! Sorry!
-- 
Celine Bursztein
------------------------------------------------------
http://celine.im
@cealtea <https://twitter.com/cealtea> on Twitter, +Celine Bursztein
<https://plus.google.com/+Celine%20Bursztein> on Google+

File: octopi.js (hunk 1 above)
Comment (eugene-eeo) [reply octopi.js:53]: Sure, are you working on it or should I merge + fix?

Comment (Cealtea) [inline octopi.js:53]: Well I did the modifications. I am going to commit!
Do you want me to do another pull request?

## 

## Celine Bursztein

http://celine.im
@cealtea https://twitter.com/cealtea on Twitter, +Celine Bursztein
https://plus.google.com/+Celine%20Bursztein on Google+

Comment (eugene-eeo) [reply octopi.js:53]: Sure, thanks + sorry for the fiasco

File: octopi.js (hunk 2 above)
Comment (Cealtea) [reply octopi.js:65]: Done!
